From 7e1f377b5bdd918dcccde7221a684c226ed1ccfe Mon Sep 17 00:00:00 2001
From: Kevin Chen <kevin_chen@aspeedtech.com>
Date: Thu, 11 Sep 2025 18:42:33 +0800
Subject: [PATCH 24/32] dts: ast2700-ibex: update reserved memory mappings

Refactor reserved memory mappings in AST2700 device tree to introduce
conditional definitions for PCIe endpoint (EP) and root complex (RC)
modes. Add support for no-map and bootph-all properties to align with
specific configurations. Include new memory regions for PCIe, eSPI,
and eDAF modes.

Change-Id: Ia519b863ad55a63109a8e9e620801f4ab3c971d8
Signed-off-by: Kevin Chen <kevin_chen@aspeedtech.com>
(cherry picked from commit b6a483776a2aa92c925bd0eaec0212529c31d25c)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi | 57 +++++++++++++++----
 arch/riscv/dts/ast2700-ibex.dts               |  5 ++
 2 files changed, 50 insertions(+), 12 deletions(-)

diff --git a/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi b/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
index 42159b3c16..0b0001f71a 100644
--- a/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
+++ b/arch/riscv/dts/ast2700-ibex-reserved-mem.dtsi
@@ -1,47 +1,80 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
-/* AST2700 reserved memories with fixed address */
-bmc_dev0_memory: bmc-dev0-memory@a3800000 {
-	reg = <0xa3800000 0x100000>;
+/* AST2700 reserved memories with no-map property */
+
+#if PCIE0_EP
+bmc_dev0_memory: bmc-dev0-memory@9b800000 {
+	reg = <0x9b800000 0x100000>;
+	no-map;
+	bootph-all;
+};
+
+xdma_memory0: xdma0 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
+	no-map;
+};
+
+pcie0_mmbi0_memory: pcie0-mmbi0-memory@9ba00000 {
+	reg = <0x9ba00000 0x200000>;
 	no-map;
 	bootph-all;
 };
+#endif
+
+#if PCIE1_EP
+bmc_dev1_memory: bmc-dev1-memory@9b900000 {
+	reg = <0x9b900000 0x100000>;
+	no-map;
+	bootph-all;
+};
+
+xdma_memory1: xdma1 {
+	size = <0x0 0x00100000>;
+	compatible = "shared-dma-pool";
+	no-map;
+};
 
-bmc_dev1_memory: bmc-dev1-memory@a3900000 {
-	reg = <0xa3900000 0x100000>;
+pcie1_mmbi4_memory: pcie1-mmbi4-memory@9bc00000 {
+	reg = <0x9bc00000 0x200000>;
 	no-map;
 	bootph-all;
 };
+#endif
 
-pcie0_mmbi0_memory: pcie0-mmbi0-memory@a3a00000 {
-	reg = <0xa3a00000 0x200000>;
+pcie2_mmbi0_memory: pcie2-mmbi0-memory@9be00000 {
+	reg = <0x9be00000 0x200000>;
 	no-map;
 	bootph-all;
 };
 
-pcie1_mmbi4_memory: pcie1-mmbi4-memory@a3c00000 {
-	reg = <0xa3c00000 0x200000>;
+espi0_mmbi_memory: espi0-mmbi-memory@9c000000 {
+	reg = <0x9c000000 0x4000000>;
 	no-map;
 	bootph-all;
 };
 
-pcie2_mmbi0_memory: pcie2-mmbi0-memory@a3e00000 {
-	reg = <0xa3e00000 0x200000>;
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
+edaf0: eadf0-memory@a0000000 {
+	reg = <0xa0000000 0x4000000>;
 	no-map;
 	bootph-all;
 };
 
-espi0_mmbi_memory: espi0-mmbi-memory@a4000000 {
+#if DUAL_NODE
+espi1_mmbi_memory: espi1-mmbi-memory@a4000000 {
 	reg = <0xa4000000 0x4000000>;
 	no-map;
 	bootph-all;
 };
 
+// This area used for eDAF DDR mode even in MIX mode or HW mode(enabled in SPL)
 edaf0: safs-ddr-mode@a8000000 {
 	reg = <0xa8000000 0x4000000>;
 	no-map;
 	bootph-all;
 };
+#endif
 
 ssp_memory: ssp-memory@ac000000 {
 	reg = <0xac000000 0x2000000>;
diff --git a/arch/riscv/dts/ast2700-ibex.dts b/arch/riscv/dts/ast2700-ibex.dts
index e777a5fa48..7553debbca 100644
--- a/arch/riscv/dts/ast2700-ibex.dts
+++ b/arch/riscv/dts/ast2700-ibex.dts
@@ -4,6 +4,11 @@
 
 #include "ast2700.dtsi"
 
+#define DUAL_NODE 0	// 1: DUAL_NODE, 0: SINGLE_NODE
+#define PCIE0_EP 1	// 1: EP, 0: RC
+#define PCIE1_EP 1	// 1: EP, 0: RC
+#define PCIE2_RC 1	// 1: RC, 0: SGMII
+
 / {
 	chosen {
 		stdout-path = &uart12;
-- 
2.34.1

